client/js/compute.js

```diff
@@ -0,0 +1,578 @@
+      sample: 'var a = ${x} split ${y}',
+      call: function(v, x, y) { return v = x.split(y) }
+    },
+    {
+      name: 'startsWith',
+      desc: 'returns true if a string x starts with a string y - case sensitive',
+      sample: 'var a = ${x} startsWith ${y}',
+      call: function(v, x, y) { return v = x.startsWith(y) }
+    },
+    {
+      name: 'toFixed',
+      desc: 'returns a formatted number based on x, using the given amount y of digits, defaulting to 0 digits',
+      sample: 'var a = ${x} toFixed ${y}',
+      call: function(v, x, y) { return v = (y !== undefined) ? x.toFixed(y) : x.toFixed() }
+    },
+    {
+      name: 'toLocaleLowerCase',
+      desc: 'returns the string x in lowercase letters, optionally according to current locale y',
+      sample: 'var a = ${x} toLocaleLowerCase ${y}',
+      call: function(v, x, y) { return v = (y !== undefined) ? x.toLocaleLowerCase(y) : x.toLocaleLowerCase() }
+    },
+    {
+      name: 'toLocaleUpperCase',
+      desc: 'returns the string in uppercase letters, optionally according to current locale y',
+      sample: 'var a = ${x} toLocaleUpperCase ${y}',
+      call: function(v, x, y) { return v = (y !== undefined) ? x.toLocaleUpperCase(y) : x.toLocaleUpperCase() }
+    },
+    {
+      name: 'replace',
+      desc: 'returns a string with the first occurence of string y replaced by string z',
+      sample: 'var a = ${x} replace ${y} ${z}',
+      call: function(v, x, y, z) { return v = x.replace(y, z) }
+    },
+    {
+      name: 'replaceAll',
+      desc: 'returns a string with all occurences of string y replaced by string z',
+      sample: 'var a = ${x} replaceAll ${y} ${z}',
+      call: function(v, x, y, z) { return v = x.replaceAll(y, z) }
+    },
+    {
+      name: 'substr',
+      desc: 'extracts and returns part of a string, starting at index y for z number of characters - or to the end (if z is zero/omitted)',
+      sample: 'var a = ${x} substr ${y} ${z}',
+      call: function(v, x, y, z) { return v = (z !== undefined) ? x.substr(y, z) : x.substr(y) }
+    },
+    {
+      name: 'getIndex',
+      desc: 'returns index y of a string or array x',
+      sample: 'var a = ${x}.${y}\nvar a = ${x} getIndex ${y}',
+      call: function(v, x, y) { return v = x[y] }
+    },
+    {
+      name: 'setIndex',
+      desc: 'sets the index x of the given variable a to the value y',
+      sample: 'var a.${x} = ${y}\nvar a = setIndex ${x} ${y}',
+      call: function(v, x, y) { v[x] = y; return v }
+    },
+    {
+      name: 'from',
+      desc: 'converts and returns an array of the given string or object',
+      sample: 'var a = from ${x}',
+      call: function(v, x) { return v = Array.from(x) }
+    },
+    {
+      name: 'isArray',
+      desc: 'returns true if the object is an array, and false if not',
+      sample: 'var a = isArray ${x}',
+      call: function(v, x) { return v = Array.isArray(x) }
+    },
+    {
+      name: 'concatArray',
```

Comment:
Why did I add this function in the first place? The code is exactly the same as `concat`, isn't it? Unrelated to the PR of course.